dali/pipeline/executor/executor2/exec2.cc

```diff
@@ -158,20 +159,27 @@ class Executor2::Impl {
     if (output_order.has_value() && output_order != ws.output_order()) {
       // Set the order of the outputs to the requested output_order - no synchronization
       // is necessary, the stream has been properly synchronized a few lines above.
+      std::unordered_set<void *> already_processed(ws.NumOutput());
```

Comment:
`void *` because `TensorList<CPUBackend>` and `TensorList<GPUBackend>` are unrelated types.

Reply:
Can we just have two separate `already_processed_cpu` and `already_processed_gpu` that are types?